### PR TITLE
Use quotes around output folder

### LINKED ISSUE #456: publish command fails when script path has spaces
System information:
```sh
λ dotnet --version
2.1.602
λ dotnet script --version
0.29.1
λ  systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.16299 N/A Build 16299
```

Replicating:
```sh
λ dotnet script new test.csx
Creating 'test.csx'
...'C:\Users\ashley\test script\test.csx' [Created]

λ dotnet script test.csx
Hello world!

λ dotnet script publish test.csx
dotnet publish failed with result '1'
```

### PR DESCRIPTION
**Pull request by seesharper**
This PR fixes #456 by making sure we put quotes around the publishing output folder

### CODE REVIEW

File: src/Dotnet.Script.Core/ScriptPublisher.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public ScriptPublisher(LogFactory logFactory, ScriptEmitter scriptEmitter)
 
             var commandRunner = new CommandRunner(logFactory);
             // todo: may want to add ability to return dotnet.exe errors
-            var exitcode = commandRunner.Execute("dotnet", $"publish \"{tempProjectPath}\" -c Release -r {runtimeIdentifier} -o {context.WorkingDirectory}");
+            var exitcode = commandRunner.Execute("dotnet", $"publish \"{tempProjectPath}\" -c Release -r {runtimeIdentifier} -o \"{context.WorkingDirectory}\"");
```
Comment (filipw) [inline src/Dotnet.Script.Core/ScriptPublisher.cs]: I think we should do the same around the nuget.config path? https://github.com/filipw/dotnet-script/blob/master/src/Dotnet.Script.DependencyModel/Context/DotnetRestorer.cs#L30

Comment (seesharper) [reply src/Dotnet.Script.Core/ScriptPublisher.cs]: Good catch. Probably also around the `packageSourcesArgument`

**Comment by seesharper** [comment]
I'm merging this then 🤘

**Comment by M-Pixel** [comment]
When will this make it onto NuGet?

**Pull request merged**